### ISSUE TITLE
Base Server Reimagining

### ISSUE THREAD

**Issue by exectails** [NoCode-NoLife/melia#207]
It's been almost eight years since we wrote the first lines of code for Melia, and we feel that there's quite a few things that could and should be improved upon before we start implementing any more content on top of a shaky base. Because of this, I'm currently in the process of revamping a couple of systems, with some tasks already done, and others still open. This issue keeps track of the things on my list.

- [x] Replace networking code with Yggdrasil
- [x] Replace command system with Yggdrasil
- [x] Replace Lua scripting with C# scripting
- [x] Add event manager for subscribing to events on the server
- [x] Rewrite properties to be easier to use and not rely on integer ids any longer
- [x] Restructure class inheritance between players, mobs, and NPCs, so they can share and also exclude code more easily
